scripts/sync_translations.py

```diff
@@ -0,0 +1,217 @@
+import configparser
+import os
+import sys
+from os.path import expanduser
+import yaml
+
+from transifex.api import transifex_api
+from transifex.api.jsonapi import exceptions
+
+NEW_PROJECT_SLUG = 'openedx-translations'
+ORGANIZATION_SLUG = 'open-edx'
+
+
+class Command:
+
+    workflow_file_path = '.github/workflows/sync-translations.yml'
+
+    def __init__(self, argv, tx_api, environ):
+        self.argv = argv
+        self.tx_api = tx_api
+        self.environ = environ
+
+    def is_dry_run(self):
+        """
+        Check if the script is running in dry-run mode.
+        """
+        return '--dry-run' in self.argv
+
+    def is_simulated_github_actions(self):
+        """
+        Check if the script is running in simulated GitHub Actions mode.
+        """
+        return '--simulate-github-workflow' in self.argv
+
+    def get_resource_url(self, resource, project_slug):
+        return f'https://www.transifex.com/{ORGANIZATION_SLUG}/{project_slug}/{resource.slug}'
+
+    def get_transifex_organization_projects(self):
+        """
+        Get openedx-translations project from Transifex.
+        """
+        tx_api_token = self.environ.get('TX_API_TOKEN')
+        if not tx_api_token:
+            config = configparser.ConfigParser()
+            config.read(expanduser('~/.transifexrc'))
+            tx_api_token = config['https://www.transifex.com']['password']
+
+        if not tx_api_token:
+            raise Exception(
+                'Error: No auth token found. '
+                'Set transifex API token via TX_API_TOKEN environment variable or via the ~/.transifexrc file.'
+            )
+
+        self.tx_api.setup(auth=tx_api_token)
+        return self.tx_api.Organization.get(slug=ORGANIZATION_SLUG).fetch('projects')
+
+    def get_resources_pair(self, new_slug, old_slug, old_project_slug):
+        """
+        Load the old and new Transifex resources pair.
+        """
+        projects = self.get_transifex_organization_projects()
+        new_project = projects.get(slug=NEW_PROJECT_SLUG)
+
+        new_resource_id = f'o:{ORGANIZATION_SLUG}:p:{new_project.slug}:r:{new_slug}'
+        print(f'new resource id: {new_resource_id}')
+        try:
+            new_resource = self.tx_api.Resource.get(id=new_resource_id)
+        except exceptions.JsonApiException as error:
+            print(f'Error: New resource error: {new_resource_id}. Error: {error}')
+            sys.exit(1)
```

Comment:
I think it's sometimes useful to include the full stack trace here.

```suggestion
            raise
```